### PR TITLE
Login logout feature

### PR DESCRIPTION
**Pull request by tayloregivens**
Updated login/logout feature

### CODE REVIEW

File: src/main/java/servlets/logServlet.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/logServlet")
+public class logServlet extends HttpServlet {
```
Comment (pavlik-y) [inline src/main/java/servlets/logServlet.java]: Log prefix is typically used for logging, not for login. Could you rename it to something related to authentication? E.g. AuthServlet.class?

Also, class names should follow UpperCamelCase (https://google.github.io/styleguide/javaguide.html#s5.2.2-class-names)

File: src/main/webapp/script.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+ function log_button(){
```
Comment (pavlik-y) [inline src/main/webapp/script.js]: This function is not attached to button, but instead typically called from "onload". It is better to give it a name that reflects its purpose, not the components where it is used. For example fetch_auth_url().

File: src/main/java/servlets/logServlet.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
```
Comment (pavlik-y) [inline src/main/java/servlets/logServlet.java]: You can make the code more readable by introducing an inner class for response format. Here is how to do it:

Outside of doGet() function create class:

```
static class AuthResponse {
  public String loginUrl;
  public String logoutUrl;
}
```
Then within doGet():
```
AuthResponse response = new AuthResponse();
...
if (userService.isUserLoggedIn()) {
  response.logoutUrl = userService.createLogoutURL("/");
} else {
  response.loginUrl = userService.createLoginURL("/");
}
response.getWriter().println(gson.toJson(response));
```

This way on the client side in log_button() function you can access fields directly:
```
if (logUrl.loginUrl) {
  console.log("received login URL");
} else {
  console.log("received logout URL");
}
```

Comment (tayloregivens) [reply src/main/java/servlets/logServlet.java]: When I tried implementing this the code became very long and I was getting error decided to leave out inner class but simply code as it is instead.

**Review by ANGELLOPARR — APPROVED** [review]
Looks good, thank you for putting in the changes suggested by Pavel!

**Pull request merged**